tests/integration_tests/test_preprocessing.py

```diff
@@ -139,3 +143,34 @@ def test_dask_known_divisions(feature_fn, csv_filename, tmpdir):
             data_df,
             skip_save_processed_input=False,
         )
+
+
+def test_number_feature_wrong_dtype(csv_filename, tmpdir):
+    """Tests that a number feature with all string values is treated as having missing values by default."""
+    data_csv_path = os.path.join(tmpdir, csv_filename)
+
+    num_feat = number_feature()
+    input_features = [num_feat]
+    output_features = [binary_feature()]
+    config = {"input_features": input_features, "output_features": output_features}
+
+    training_data_csv_path = generate_data(input_features, output_features, data_csv_path)
+    df = pd.read_csv(training_data_csv_path)
+
+    # convert numbers to random strings
```

Comment:
nit: optional, instead of defining a new random string function, it seems cleaner to use a synthetic text input feature and then only change the input feature type just in the config, like config["input_features"][0]["type"] = "number"